### PR TITLE
Add mysql-uqi-client and mysql-demo workflow

### PR DESCRIPTION
**Pull request by jonmagic**
## Why?

Closes https://github.com/open-truss/open-truss/issues/42

## How?

We wrote a uqi wrapper for the mysql2 client and tested it by adding a mysql-demo workflow.

```yaml
workflow:
  version: 1
  frames:
    - frame:
      data:
        query: select * from sys.metrics
        source: mysql-demo
      view:
        component: TrinoDemo
```

CleanShot 2024-02-27 at 14 59 49@2x

## Collaborators

- @hktouw
- @spicycode
- @tmetz

### CODE REVIEW

File: demo-app/src/lib/mysql-uqi-client.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+} from '@open-truss/open-truss'
+import mysql, { type Connection, type FieldPacket } from 'mysql2/promise'
+import { URL } from 'url'
+
+export interface MysqlConfig {
+  uri: string
+  socketPath?: string
+}
+
+function makeUqiColumnCompatible(fields: FieldPacket[]): UqiColumn[] {
```
Comment (kmcq) [inline demo-app/src/lib/mysql-uqi-client.ts:18]: Maybe a comment here?

```suggestion
// mysql2's FieldPacket.type is a number but UqiColumn.type is a string so we map
// the number to a string.
function makeUqiColumnCompatible(fields: FieldPacket[]): UqiColumn[] {
```

File: demo-app/src/lib/mysql-uqi-client.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      name: String(field.name),
+      type: String(field.type),
+    }
+  })
+}
+
+export default async function (config: MysqlConfig): Promise<UqiClient> {
+  const parsedUrl = new URL(config.uri)
+
+  const typeMappings: Record<string, UqiMappedType> = {
```
Comment (kmcq) [inline demo-app/src/lib/mysql-uqi-client.ts:30]: Would it work to have this a constant? might make this function a bit easier to read

**Review by kmcq — APPROVED** [review]
Looks great! I couldn't get mysql set up but it looks legit :)

**Review by hktouw — APPROVED** [review]
This looks legit to me. Nice work!

I didn't want to bother setting up another mysql locally, but modifying the mysql creds to point to my already existing instance seemed to work just fine.

‎localhost:3000otplayground7-mysql-demo 2024-02-29 at 6 09 30 PM

**Pull request merged**